cogs/team.py

```diff
@@ -0,0 +1,205 @@
+        # if none, delete team and respective channels
+        # also delete the role
+        team_members = await team_query.get_team_members(team_name)
+        if team_members:
+            await interaction.response.send_message(
+                "You have left the team.", ephemeral=True
+            )
+            return
+
+        await interaction.response.send_message(
+            "You have left the team. Since there are no members left in the team, the channels will be deleted.",
+            ephemeral=True,
+        )
+
+        # if here, then there are no members remaining in the teams
+        text_channel = guild.get_channel(int(team.text_channel_id))
+        voice_channel = guild.get_channel(int(team.voice_channel_id))
+        category_channel = guild.get_channel(int(team.category_channel_id))
+
+        await text_channel.delete()
+        await voice_channel.delete()
+        await category_channel.delete()
+        await role.delete()
+
+        # also delete the team
+        await team_query.remove_team(team_name)
+
+    @app_commands.command(name="invite")
+    async def invite(
+        self, interaction: discord.Interaction, invited_user: discord.Member
+    ):
+        user = interaction.user
+        guild = interaction.guild
+
+        # check user is already in a team
+        player = await player_query.get_player(str(user.id))
+        if not player:
+            await interaction.response.send_message(
+                "You must be in a team to use this command.", ephemeral=True
+            )
+            return
+
+        team_name = player.team_name
+
+        # check invited user is not already in a team
+        if await player_query.get_player(str(invited_user.id)):
+            await interaction.response.send_message(
+                "The user you're trying to invite is already in a team.", ephemeral=True
+            )
+            return
+
+        # check that team is not full
+        if len(await team_query.get_team_members()) == MAX_TEAM_SIZE:
+            await interaction.response.send_message(
+                "Your team is full.", ephemeral=True
+            )
+
+        # otherwise send an invite
+        embed = discord.Embed(
+            title=f"{team_name} Invitation",
+            description=f"{user.name} has invited you to their team!",
+        )
+
+        accept_btn = discord.ui.Button(label="Accept", style=discord.ButtonStyle.green)
+        reject_btn = discord.ui.Button(label="Reject", style=discord.ButtonStyle.red)
+
+        view = discord.ui.View()
+        view.add_item(accept_btn)
+        view.add_item(reject_btn)
+
+        """
+        Completely optional but we could make it so players can add
+        custom accept/reject messages.
+        """
+
+        # add invited user to the team and database
+        async def accept_callback(interaction: discord.Interaction):
+            new_player = interaction.user
+            team = await team_query.get_team(team_name)
```

Comment:
what happens if the team is removed after sending the invite, but before the user accept it?

Reply:
additionally, what if there are two or more invites send when say, there are 4 current people in a team? wouldn't it allow for more than 5 people in a team?